storage/integration_test.go

```diff
@@ -4404,74 +4389,103 @@ func TestIntegration_Scopes(t *testing.T) {
 
 }
 
-func TestIntegration_SignedURL_Bucket(t *testing.T) {
+func TestIntegration_SignedURL_WithCreds(t *testing.T) {
 	ctx := context.Background()
 
-	if testing.Short() && !replaying {
-		t.Skip("Integration tests skipped in short mode")
+	// By default we are authed with a token source, so don't have the context to
+	// read some of the fields from the keyfile.
+	// Here we explicitly send the key to the client.
+	creds, err := findTestCredentials(ctx, "GCLOUD_TESTS_GOLANG_KEY", ScopeFullControl, "https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		t.Fatalf("unable to find test credentials: %v", err)
 	}
 
-	// We explictly send the key to the client to sign with the private key
-	clientWithCredentials := newTestClientWithExplicitCredentials(ctx, t)
-	defer clientWithCredentials.Close()
+	multiTransportTest(skipGRPC("creds capture logic must be implemented for gRPC constructor"),
+		t, func(t *testing.T, ctx context.Context, bucket, _ string, clientWithCredentials *Client) {
+			h := testHelper{t}
+			obj := uidSpace.New()
+			o := clientWithCredentials.Bucket(bucket).Object(obj)
+			contents := []byte("test")
+			if err := writeObject(ctx, o, "text/plain", contents); err != nil {
+				t.Fatalf("writing: %v", err)
+			}
+			defer h.mustDeleteObject(o)
+
+			for _, test := range []struct {
+				desc   string
+				opts   SignedURLOptions
+				client *Client
+			}{
+				{
+					desc: "signing with the private key",
+					opts: SignedURLOptions{
+						Method:  "GET",
+						Expires: time.Now().Add(30 * time.Second),
+					},
+					client: clientWithCredentials,
+				},
+			} {
+				t.Run(test.desc, func(t *testing.T) {
+					bkt := test.client.Bucket(bucket)
+					url, err := bkt.SignedURL(obj, &test.opts)
+					if err != nil {
+						t.Fatalf("unable to create signed URL: %v", err)
+					}
 
-	// Create another client to test the sign byte function as well
+					if err := verifySignedURL(url, nil, contents); err != nil {
+						t.Fatalf("problem with the signed URL: %v", err)
+					}
+				})
+			}
+		}, option.WithCredentials(creds))
+}
 
-	scopes := []string{ScopeFullControl, "https://www.googleapis.com/auth/cloud-platform"}
-	ts := testutil.TokenSource(ctx, scopes...)
-	if ts == nil {
-		t.Fatalf("Cannot get token source to create client")
-	}
-	clientWithoutPrivateKey := testConfig(ctx, t, option.WithTokenSource(ts))
-	defer clientWithoutPrivateKey.Close()
+func TestIntegration_SignedURL_Default(t *testing.T) {
```

Comment:
This only tests the "default" under certain auth conditions - I would update this name to be more specific, maybe something like:

```suggestion
func TestIntegration_SignedURL_DefaultSignBytes(t *testing.T) {
```